packages/datagateway-table/src/breadcrumbs.component.tsx

```diff
@@ -0,0 +1,538 @@
+          }
+        }
+        // Ensure that we store the last path in a separate field and that it is not the entity name.
+      } else if (index === pathLength - 1) {
+        if (entity !== updatedState.base.entityName) {
+          // Update the state to say that e.g. if path is /browse/investigation/,
+          // then display name would be just Browse > *Investigations*.
+          // e.g. Browse > Investigations > *Proposal 1* (Datasets) etc.
+          updatedState = {
+            ...updatedState,
+            last: {
+              displayName:
+                `${entity}`.charAt(0).toUpperCase() + `${entity}s`.slice(1),
+            },
+          };
+        } else {
+          // If the base is the current top directory, then update the last field.
+          updatedState = {
+            ...updatedState,
+            base: {
+              ...updatedState.base,
+
+              isLast: true,
+            },
+          };
+        }
+      }
+
+      // Increment the hierarchy count for the next entity (if there is one).
+      hierarchyCount++;
+    }
+
+    // Update the final state.
+    this.setState(updatedState);
+    this.isBreadcrumbUpdated = true;
+  };
+
+  private getEntityInformation = async (
+    requestEntityUrl: string,
+    entityField: string
+  ): Promise<string> => {
+    let entityName = '';
+    const requestUrl = `${this.props.apiUrl}/${requestEntityUrl}`;
+
+    // Make a GET request to the specified URL.
+    entityName = await axios
+      .get(requestUrl, {
+        headers: {
+          Authorization: `Bearer ${window.localStorage.getItem('daaas:token')}`,
+        },
+      })
+      .then(response => {
+        // Return the property in the data received.
+        return response.data[entityField];
+      });
+
+    return entityName;
+  };
+
+  public render(): React.ReactElement {
+    // Filter to ensure the breadcrumbs are only for this path
+    // (in the event the user moved back and changed location using the breadcrumb).
+    const breadcrumbState = {
+      ...this.state,
+      currentHierarchy: this.state.currentHierarchy.filter(
+        (value: Breadcrumb, index: number) => {
+          // Return the breadcrumbs with pathnames in our current location and with the related IDs.
+          // The index of the ID related to the path name is derived from the path name index add one
+          // to give the ID generally and then shifted by two as this occurs every two elements.
+          return (
+            this.currentPathnames.includes(value.pathName) &&
+            this.currentPathnames[(index + 1) * 2] === value.id
+          );
+        }
+      ),
+    };
+
+    const hierarchyKeys = Object.keys(breadcrumbState.currentHierarchy);
+    return (
+      <div>
+        <Paper elevation={0}>
+          <Route>
+            {/* // Ensure that there is a path to render, otherwise do not show any breadcrumb. */}
+            {this.currentPathnames.length > 0 ? (
+              <Breadcrumbs
```

Comment:
It's not totally obvious that the base breadcrumb is part of a breadcrumb trail
- Have an overall background?
- Have a base entity/icon
- Background colour on the breadcrumb